src/tools/remote/src/main/java/com/google/devtools/build/remote/worker/OnDiskBlobStoreCache.java

```diff
@@ -92,6 +103,29 @@ public void downloadTree(
     }
   }
 
+  @Override
+  public ListenableFuture<byte[]> downloadBlob(RemoteActionExecutionContext context, Digest digest) {
+    logger.atFine().log("downloadBlob digest=%s invocationId=%s",
+        digest.getHash(), context.getRequestMetadata().getToolInvocationId());
+
+    if (remoteWorkerOptions.fakeErrorForDuplicatedDownloads) {
+      // Only populate numberOfDownloadsPerDigestAndInvocation when fakeErrorForDuplicatedDownloads is enabled
+      // to avoid unnecessary unbounded memory growth.
+      String uniqueDownloadKey = String.format("%s_%s",
+          digest.getHash(), context.getRequestMetadata().getToolInvocationId());
+      int numberOfDownloads = numberOfDownloadsPerDigestAndInvocation.computeIfAbsent(
+          uniqueDownloadKey,
+          d -> new AtomicInteger()).incrementAndGet();
```

Comment:
You can use a plain `Integer` for the values; `computeIfAbsent` already handles concurrent updates to the same key.

Reply:
Right, I meant to say to use `compute`, but `merge` works too (and is arguably better).